### PR TITLE
Fixes optimizer-runs-raven bug

### LINKED ISSUE #606: [rework] Optimizer and point-probability
In RAVEN running RAVEN, when the outer RAVEN reads a data obj from inner RAVEN that was created by a Optimizer, the point-probability fails as follows:

```
WARNING: XML node "PointSet" != param spec name "DataSet"
<DataObjects.XDataObject.MessageCourier instance at 0x1148d7950> <DataObjects.XPointSet.PointSet object at 0x1148a4610> Reading data from "/Users/epinas/projects/hybrid/CashFlowCalcs/2017September_INLEXT-17-42441/Opt_C_NoModelica_D_TEST/./OPTrun/RrR++0_0_0/./Back_to_MASTER.csv.csv" DEBUG debug None
<DataObjects.XDataObject.MessageCourier instance at 0x1148d7950> <DataObjects.XPointSet.PointSet object at 0x1148a4610> Reading metadata from "/Users/epinas/projects/hybrid/CashFlowCalcs/2017September_INLEXT-17-42441/Opt_C_NoModelica_D_TEST/./OPTrun/RrR++0_0_0/./Back_to_MASTER.xml" DEBUG debug None
Exception in thread RrR++0_0_0:
Traceback (most recent call last):
  File "/Users/epinas/.conda/envs/raven_libraries/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/Users/epinas/.conda/envs/raven_libraries/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/epinas/projects/raven/framework/Runners/SharedMemoryRunner.py", line 135, in <lambda>
    self.thread = threading.Thread(target = lambda q, *arg : q.append(self.functionToRun(*arg)), name = self.identifier, args=(self.subque,)+tuple(self.args))
  File "/Users/epinas/projects/raven/framework/Models/Code.py", line 573, in evaluateSample
    rlz['PointProbability'] = np.atleast_1d(rlz.get('PointProbability',1.0) * kwargs.get('PointProbability',1.0))
TypeError: unsupported operand type(s) for *: 'float' and 'NoneType'
```

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
Closes #606 by closing some errors not seen unless an optimizer runs a code (specifically code RAVEN), and adds a regression test to check for it.

### CODE REVIEW

File: tests/framework/Optimizers/RavenRunningRaven/innerRun/parabola.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#***************************************
+#* Simple analytic test ExternalModule *
+#***************************************
+#
```
Comment (alfoa) [inline tests/framework/Optimizers/RavenRunningRaven/innerRun/parabola.py]: can't we just point to the model in /tests/framework/analyticalModels?

File: tests/framework/Optimizers/RavenRunningRaven/inner_opt.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        <lowerBound>-2</lowerBound>
+        <initial>-0.75</initial>
+      </variable>
+      <constant name="center">1</constant>
+      <objectVar>reg</objectVar>
+    </SPSA>
+  </Optimizers>
+
+  <Models>
+    <ExternalModel ModuleToLoad="parabola" name="parabola" subType="">
```
Comment (alfoa) [inline tests/framework/Optimizers/RavenRunningRaven/inner_opt.xml]: ```../../framework/AnalyticModels/parabola```? so we do not copy another file?

File: tests/framework/Optimizers/RavenRunningRaven/innerRun/parabola.py (hunk 1 above)
Comment (PaulTalbot-INL) [reply tests/framework/Optimizers/RavenRunningRaven/innerRun/parabola.py]: If it were the same model, sure, but it's not.

File: tests/framework/Optimizers/RavenRunningRaven/inner_opt.xml (hunk 2 above)
Comment (PaulTalbot-INL) [reply tests/framework/Optimizers/RavenRunningRaven/inner_opt.xml]: If it were the same model, sure, but it's not.

**Comment by PaulTalbot-INL** [comment]
Yes, I have to fix that still, but it's moot until #590 is merged.

**Comment by PaulTalbot-INL** [comment]
Tests passed except mac, who is pending library setup fixes in CIVET.

**Pull request merged**